stm32-modules/include/heater-shaker/heater-shaker/motor_task.hpp

```diff
@@ -167,18 +169,37 @@ class MotorTask {
             policy.homing_solenoid_disengage();
             auto error = policy.set_rpm(msg.target_rpm);
             state.status = State::RUNNING;
-            auto response = messages::AcknowledgePrevious{
-                .responding_to_id = msg.id, .with_error = error};
-            if (msg.from_system) {
-                static_cast<void>(
-                    task_registry->system->get_message_queue().try_send(
-                        messages::SystemMessage(response)));
+            policy.delay_ticks(MOTOR_START_WAIT_TICKS);
+            static_cast<void>(
+                get_message_queue().try_send(messages::CheckMotorStartMessage{
```

Comment:
I'm not clear on why this is an extra message. Since the handling for this message includes the necessary delay, couldn't the error handling be done in the same function?